src/java/main/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -479,7 +479,7 @@ private boolean checkFourLetterWord(final SelectionKey k, final int len)
         // We take advantage of the limited size of the length to look
         // for cmds. They are all 4-bytes which fits inside of an int
         String cmd = FourLetterCommands.getCmdMapView().get(len);
-        if (cmd == null) {
+        if (cmd == null || !FourLetterCommands.getWhiteListedCmdView().contains(cmd)) {
```

Comment:
if request is for 4lw command, it should be processed in that way only. If false is returned from here, the request will proceed as the normal request.
This is major issue in the current patch